### DISCUSSION TITLE
Add an image editing option to the Image field in an item of any collection

### DISCUSSION THREAD

**Discussion by qborisb** [directus/directus#14395] [Feature Requests]
We would find it very useful to be able to edit images right from the item settings in a collection:

![Screenshot 2022-07-12 at 12 16 26](https://user-images.githubusercontent.com/54835674/178468023-aea8490f-e508-40b7-9ed4-03d0ac5cfe0a.png)

![Screenshot 2022-07-12 at 12 16 52](https://user-images.githubusercontent.com/54835674/178468089-58265341-d4aa-4b74-955b-619d34dd6762.png)

Seems like it is not difficult to implement, `image-editor.vue` modal that is being used in file editing can also be used here.

We are happy to create a PR for this!

**Comment by rijkvanzanten** [comment]
@qborisb Mind reviewing https://github.com/directus/directus/pull/10488? 🙂

**Comment by qborisb** [comment]
@rijkvanzanten yes, sure :)

**Comment by qborisb** [comment]
@rijkvanzanten Additionally, we thought that it would be nice to to have **custom aspect ratios** in the image editing modal (as we want to have other ratios available for our users), for example `16:1` and `5:1` instead of the default ones:

![Screenshot 2022-07-12 at 18 11 39](https://user-images.githubusercontent.com/54835674/178541433-9e08eb48-8005-4814-b3a8-90e565d2b848.png)

This would involve adding a prop to the `image-editor.vue`, something like `custom-aspect-ratios`, that, if passed, would overwrite the default aspect ratios (16:9, square, etc.). These `custom-aspect-ratios` can be an option in the settings of an Image interface:

![Screenshot 2022-07-12 at 18 08 59](https://user-images.githubusercontent.com/54835674/178540916-03d23402-6d6c-42f2-9934-6891cd9080f8.png)

This would allow more customisability with the cropping options, while not changing the behaviour for the current users. People willing to have custom aspect ratios would need to just specify them in the `Data Model -> Settings of the image field -> image interface settings`, and these aspect ratios would be used instead of the default ones.

If you accept this feature request, we are happy to collaborate with @juancarlosjr97 (if he agrees) on his PR (#10488) to implement both features together. Or, we could create a separate PR for that :)

**Reply by juancarlosjr97** [reply]
Awesome ideas @qborisb, I like them :) 

Sure, lets work together, I have added you to my fork repository, but up to you if you want to fork my fork or copy the changes, as I do not have a lot of time in the upcoming weeks to sit and work on it and coordinate with you and the Directus team.

**Reply by janwillemvd** [reply]
Thanks @juancarlosjr97. I'm a colleague of @qborisb and Im' sure @qborisb  is happy to work on this next week :)

**Reply by rijkvanzanten** [reply]
Lets get #10488 merged first, as it should be ready to go, and tackle this enhancement as a separate effort 🙂

**Reply by qborisb** [reply]
@rijkvanzanten here is my implementation of the feature #14510

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
